### DISCUSSION TITLE
Support for UDFs that take variadic arrays

### DISCUSSION THREAD

**Discussion by jcuenod** [duckdb/duckdb#9683] [Q&A] [answered]
I want to register a UDF that takes an arbitrary number of UBIGINT arrays. It seems as though this is not yet possible, am I correct?

**Comment by Tishj** [comment]
What language?

**Reply by jcuenod** [reply]
Ideally node, but that's just proving out the idea.

**Reply by Tishj** [reply]
we have VARARGS, that's how we implement variadic arguments
It could be that the node client doesn't support this?

The Node client has its own repository, maybe make an issue there?

**Reply by Tishj** [reply, marked as answer]
For example, this is internally the prototype for `map_from_entries`
```c++
ScalarFunction MapFromEntriesFun::GetFunction() {
	//! the arguments and return types are actually set in the binder function
	ScalarFunction fun({}, LogicalTypeId::MAP, MapFromEntriesFunction, MapFromEntriesBind);
	fun.null_handling = FunctionNullHandling::DEFAULT_NULL_HANDLING;
	fun.varargs = LogicalType::ANY;
	return fun;
}
```

In the binder code we check which types we've received and throw an error if they are not accepted

**Reply by jcuenod** [reply]
@Tishj does the python library support this?

**Reply by Tishj** [reply]
More or less
```py
    def test_varargs(self):
        def variable_args(*args):
            amount = len(args)
            return amount

        con = duckdb.connect()
        con.create_function('varargs', variable_args, None, BIGINT)
        res = con.sql("""select varargs('5', '3', '2', 1, 0.12345)""").fetchall()
        assert res == [(5,)]
```

**Comment by jcuenod** [comment]
Awesome, got it working, thanks!
